users/user-guide-query/multi-stage-query/operator-types/aggregate.md

```diff
@@ -0,0 +1,194 @@
+---
+description: >-
+  Describes the aggregate relation operator in the multi-stage query engine.
+---
+
+# Aggregate relational operator
+
+The aggregate operator is used to perform calculations on a set of rows and return a single row of results.
+
+This page describes the aggregate operator defined in the relational algebra used by multi-stage queries.
+This operator is generated by the multi-stage query engine when you _aggregate functions_ in a query either with or
+without a `group by` clause.
+
+## Implementation details
+Aggregate operations may be expensive in terms of memory, CPU and network usage. 
+As explained in [understanding stages](../understanding-stages.md), the multi-stage query engine breaks down the query 
+into multiple stages and each stage is then executed in parallel on different _workers_.
+Each worker processes a subset of the data and sends the results to the _coordinator_ which then aggregates the results.
+When possible, the multi-stage query engine will try to apply a divide-and-conquer strategy to reduce the amount of data
+that needs to be processed in the _coordinator_ stage.
+
+For example if the aggregation function is a sum, the engine will try to sum the results of each worker before sending
+the partial result to the _coordinator_, which would then sum the partial results in order to get the final result.
+But some aggregation functions, like `count(distinct)`, cannot be computed in this way and require all the data to be
+processed in the _coordinator_ stage.
+
+In Apache Pinot 1.1.0, the multi-stage query engine always keeps the data in memory.
+This means that the amount of memory used by the engine is proportional to the number of groups generated by the
+`group by` clause and the amount of data that needs to be kept for each group (which depends on the aggregation 
+function).
+
+Even when the aggregation function is a simple `count`, which only requires to keep a long for each group in memory,
+the amount of memory used can be high if the number of groups is high.
+This is why the engine limits the number of groups.
+By default, this limit is 100.000, but this can be changed by providing hints.
+
+### Blocking nature
+The aggregate operator is a blocking operator. It needs to consume all the input data before emitting the result.
+
+## Hints
+
+### num_groups_limit
+Type: Integer
+
+Default: 100.000
+
+Defines the max number of groups that can be created by the `group by` clause. 
+If the number of groups exceeds this limit, the query will not fail but will stop the execution.
```

Comment:
the query will be marked as partial, the stat will say that the limit has been reached in this operator.

About the specific behavior, it isn't clear to me as well. A fast read on [AggregateOperator](https://github.com/apache/pinot/blob/d00870906412506fbd03f58fb5f4b78022b2b43d/pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java) seems to point that we are actually not aborting the execution:

```java
  @Override
  protected TransferableBlock getNextBlock() {
    if (_hasConstructedAggregateBlock) {
      assert _eosBlock != null;
      return _eosBlock;
    }
    TransferableBlock finalBlock = _aggregationExecutor != null ? consumeAggregation() : consumeGroupBy(); // neither of these checks _groupByExecutor.isNumGroupsLimitReached()
    // returning upstream error block if finalBlock contains error.
    if (finalBlock.isErrorBlock()) {
      return finalBlock;
    }
    assert finalBlock.isSuccessfulEndOfStreamBlock() : "Final block must be EOS block";
    _eosBlock = updateEosBlock(finalBlock, _statMap);
    return produceAggregatedBlock();  // this updates the stats and calls _input.earlyTerminate(); if needed
  }
```

So I think this may be a bug. In theory what it should do is to is to fetch all blocks from the input and calculate the value for each one of the first 100k groups found but ignore the other.

cc @npawar